### PR TITLE
Support submodules in GitHub workflows

### PR DESCRIPTION
**Pull request by AlD**

### CODE REVIEW

**Comment by AlD** [comment]
Depends on:
* https://github.com/quassel/quassel-build-env/pull/1
* https://github.com/quassel/quassel-build-env/pull/2

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -271,7 +277,11 @@ jobs:
       run: |
         mkdir assets
         TAGNAME=$(git rev-parse --abbrev-ref $GITHUB_REF)
-        git archive --format=tar --prefix=quassel-$TAGNAME/ $TAGNAME | xz > assets/quassel-$TAGNAME.tar.xz
+        PREFIX=quassel-$TAGNAME/
+        TAR_BASE=assets/quassel-$TAGNAME
+        git archive --format=tar --prefix=$PREFIX -o $TAR_BASE.tar $TAGNAME
+        git submodule foreach --recursive "git archive --prefix=$PREFIX -o $TAR_BASE-\$sha1.tar HEAD"
+        tar --concatenate --file=$TAR_BASE.tar $TAR_BASE-*.tar && rm $TAR_BASE-*.tar
```
Comment (Sput42) [inline .github/workflows/main.yml:284]: This lacks the compression part, no?

**Comment by Sput42** [comment]
I have merged the dependencies, however it turns out that Docker no longer automatically builds new images due to their recent license changes. I have applied to have Quassel accepted as a FOSS project in the hopes that we get that functionality back.

File: .github/workflows/main.yml (hunk 1 above)
Comment (AlD) [reply .github/workflows/main.yml]: True, fixed.
Also added a workaround for https://lists.gnu.org/archive/html/bug-tar/2008-08/msg00002.html.

**Comment by Sput42** [comment]
recheck

**Comment by Sput42** [comment]
@AlD  Since Docker keeps ignoring my request, I have now found out how to update the images manually, and have done so. As I don't seem to be able to re-run the checks via the webinterface (wtf), could you please do some force-push so we can see if it works this time?

**Review by Sput42 — APPROVED** [review]

**Pull request merged**